cookbooks/bach_spark/recipes/spark_package.rb

```diff
@@ -0,0 +1,36 @@
+spark_tar_file = node[:spark][:download][:file]
+spark_tar_url = node[:spark][:download][:url]
+spark_download_dir = node[:spark][:download][:dir]
+spark_install_dir = node[:spark][:package][:base]
+spark_pkg_prefix = node[:spark][:package][:prefix]
+spark_pkg_version = node[:spark][:package][:version]
+
+gem_package "fpm" do
+   action :nothing
+end.run_action(:install)
+
+remote_file "#{spark_download_dir}/#{spark_tar_file}" do
+  source "#{spark_tar_url}/#{spark_tar_file}"
+  not_if { File.exists?("#{spark_download_dir}/#{spark_tar_file}") }
+end
+
+bash "build_spark_package" do
+  cwd spark_download_dir
+  user 'root'
+  group 'root'
+  code %Q{
+    fpm -s tar -t deb --prefix #{spark_install_dir} -n #{spark_pkg_prefix} -v #{spark_pkg_version} --description "Spark Package with Hadoop 2.6" #{spark_tar_file}
```

Comment:
I would not mention Hadoop 2.6 since we have nothing which is 2.6 dependent and this will likely become stale.